### ISSUE TITLE
Evaluate maintenance workflow on new tools

### ISSUE THREAD

**Issue by jberry-suse** [openSUSE/openSUSE-release-tools#1662]
Deploy the three primary tools:

- [x] `factory-auto`: no comment only mode, just add as reviewer (low risk)
- [x] `leaper`: comment only mode for `openSUSE:Maintenance` project (likely needs tweaks)
- [x] `repo-checker`: ~comment only mode~ as reviewer on `openSUSE:Leap:15.0:Update` requests
- [x] `repo-checker`: enable `project_only` run on `openSUSE:Leap:15.0:Update` (requires write to `00Meta` package)

Those from maintenance should be notified of this issue and report feedback here before requiring `leaper` and `repo-checker`. @lnussel can you add anyone you feel is relevant or perhaps send mail?

After the remaining PR in this milestone is merged (tomorrow presumably) it is just a matter of updating the tools on `packagelists` and `tortuga` then enabling the relevant services.

Perhaps while we are starting fresh, `factory-auto` user instead of group can be added to `openSUSE:Maintenance` (per #1181)? I'll add a service file to run as user and both can be run until _Leap_ and _Factory_ follow suit.

**Comment by jberry-suse** [comment]
I remembered incorrectly, maintenance sub-package and service was created, as such those will need to be setup elsewhere to run during transition.

**Comment by jberry-suse** [comment]
Either that or I pull the removal out as separate PR and accept that after completing the switch.

**Comment by jberry-suse** [comment]
@lnussel the `repo-checker` user needs write permission to `openSUSE:Leap:15.0:Update/00Meta`.

**Comment by jberry-suse** [comment]
Included `Leap:42.3:Update` in list. Based on my testing results I would suggest we go for hard deployment of `factory-auto` and `repo-checker`. `leaper` run properly on all currently open requests, but need to confirm basically expected maintenance to change origin file to _fork_ half the time.

**Comment by jberry-suse** [comment]
See [gist](https://gist.github.com/jberry-suse/8f36c54bf8e81af2cb10bdd4764c4352) for complete test run output.

**Comment by lnussel** [comment]
yeah, I guess the origin change is mostly expected, except for packages coming from SLE.

**Comment by jberry-suse** [comment]
Updated tortuga (decided against 42.3 project_only since it has known issues and will send out comments).

Running `20180823.db50e79` and TW `20180812`.

**Comment by jberry-suse** [comment]
Cannot connect to packagelists vm for more than a couple seconds before drop. That will need to be resolved before the remaining items are deployed.

**Comment by jberry-suse** [comment]
Accepted [first live request](https://build.opensuse.org/request/show/631127)

```
ug 23 16:59:53 repochecker osrt-repo-checker[13433]: [I] checking 631127
Aug 23 16:59:53 repochecker osrt-repo-checker[13433]: [D] skipping review of action targeting openSUSE:Backports:SLE-12
Aug 23 16:59:53 repochecker osrt-repo-checker[13433]: [D] skipping review of action targeting openSUSE:Backports:SLE-15:Update
Aug 23 17:00:01 repochecker osrt-repo-checker[13433]: [I] checking openSUSE:Maintenance:8648/openSUSE_Leap_15.0_Update@8390572[3]
Aug 23 17:00:06 repochecker osrt-repo-checker[13433]: [I] mirroring openSUSE:Maintenance:8648/openSUSE_Leap_15.0_Update/x86_64
Aug 23 17:00:07 repochecker osrt-repo-checker[13433]: downloading 1 new packages
Aug 23 17:00:07 repochecker osrt-repo-checker[13433]: [I] mirroring openSUSE:Leap:15.0:Update/standard/x86_64
Aug 23 17:00:07 repochecker osrt-repo-checker[13433]: [I] mirroring openSUSE:Leap:15.0/standard/x86_64
Aug 23 17:00:16 repochecker osrt-repo-checker[13433]: [I] cycle check: start
Aug 23 17:00:16 repochecker osrt-repo-checker[13433]: [I] cycle check: passed
Aug 23 17:00:16 repochecker osrt-repo-checker[13433]: [I] install check: start (ignore:True, whitelist:13, parse:False, no_filter:False)
Aug 23 17:06:02 repochecker osrt-repo-checker[13433]: [I] install check: passed
Aug 23 17:06:02 repochecker osrt-repo-checker[13433]: [D] broadening search to include any state on openSUSE:Maintenance:8648
Aug 23 17:06:02 repochecker osrt-repo-checker[13433]: [D] no previous comment to replace on openSUSE:Maintenance:8648
Aug 23 17:06:09 repochecker osrt-repo-checker[13433]: [I] checking openSUSE:Maintenance:8648/openSUSE_Leap_42.3_Update@992cb54[3]
Aug 23 17:06:14 repochecker osrt-repo-checker[13433]: [I] mirroring openSUSE:Maintenance:8648/openSUSE_Leap_42.3_Update/x86_64
Aug 23 17:06:14 repochecker osrt-repo-checker[13433]: downloading 1 new packages
Aug 23 17:06:14 repochecker osrt-repo-checker[13433]: [I] mirroring openSUSE:Leap:42.3:Update/standard/x86_64
Aug 23 17:09:59 repochecker osrt-repo-checker[13433]: downloading 3982 new packages
Aug 23 17:09:59 repochecker osrt-repo-checker[13433]: [I] mirroring openSUSE:Leap:42.3/standard/x86_64
Aug 23 17:37:10 repochecker osrt-repo-checker[13433]: downloading 33615 new packages
Aug 23 17:37:10 repochecker osrt-repo-checker[13433]: [W] no project_only run from which to extract existing problems
Aug 23 17:37:10 repochecker osrt-repo-checker[13433]: [I] cycle check: start
Aug 23 17:37:10 repochecker osrt-repo-checker[13433]: [I] cycle check: passed
Aug 23 17:37:10 repochecker osrt-repo-checker[13433]: [I] install check: start (ignore:True, whitelist:0, parse:False, no_filter:False)
Aug 23 17:38:10 repochecker osrt-repo-checker[13433]: [I] install check: passed
Aug 23 17:38:10 repochecker osrt-repo-checker[13433]: [D] broadening search to include any state on openSUSE:Maintenance:8648
Aug 23 17:38:10 repochecker osrt-repo-checker[13433]: [D] no previous comment to replace on openSUSE:Maintenance:8648
Aug 23 17:38:10 repochecker osrt-repo-checker[13433]: [D] skipping review of action targeting openSUSE:Backports:SLE-12
Aug 23 17:38:10 repochecker osrt-repo-checker[13433]: [D] skipping review of action targeting openSUSE:Backports:SLE-15:Update
Aug 23 17:38:24 repochecker osrt-repo-checker[13433]: [I] 631127 accepted: cycle and install check passed
Aug 23 17:38:24 repochecker osrt-repo-checker[13433]: [D] setting 631127 to accepted
```

**Comment by jberry-suse** [comment]
packagelists: updated openSUSE-release-tools packages to 20180823.b2be0e2

Verified new features worked:

```
osrt-check-source@packagelists:~> osrt-check_source --debug --dry id 630848
[I] checking 630848
[I] home:AndreasStieger:branches:OBS_Maintained:libqt5-qtbase/libqt5-qtbase.openSUSE_Leap_15.0_Update@5b2bd0df00f370db1fe0023b7b0b152e -> openSUSE:Leap:15.0:Update/libqt5-qtbase
[I] 630848 accepted: Check script succeeded
[D] 630848 review not changed
```

**Comment by jberry-suse** [comment]
Verified leaper:

```
osrt-leaper@packagelists:~> osrt-leaper --debug --dry id 630848
[I] checking 630848
[I] home:AndreasStieger:branches:OBS_Maintained:libqt5-qtbase/libqt5-qtbase.openSUSE_Leap_15.0_Update@5b2bd0df00f370db1fe0023b7b0b152e -> openSUSE:Leap:15.0:Update/libqt5-qtbase
[I] expected origin is 'SUSE:SLE-15:GA' (changed)
[D] checking libqt5-qtbase in SUSE:SLE-15:GA
[D] 1813b64f30c88885258d57d2202f593b not the latest version, checking history
[D] checking ef24c391293f526554859fee82d8938c
[D] checking 49348531970c19f8619f8a8b43e9e227
[D] checking 26a0c3c0341b3249aab09c9fce4ed609
[D] checking b1aba294782352bec2a5d4679a76bdba
[D] checking 3ce2ccf5f181e6144dcb873ed77a338c
[D] srcmd5 not found in history either
[D] checking requests
[W] no devel project found for openSUSE:Leap:15.0:Update/libqt5-qtbase
[D] review result: None
[D] broadening search to include any state on 630848
[D] adding comment to 630848: <!-- Leaper state=seen result=None -->

home:AndreasStieger:branches:OBS_Maintained:libqt5-qtbase/libqt5-qtbase.openSUSE_Leap_15.0_Update@5b2bd0df00f370db1fe0023b7b0b152e -> openSUSE:Leap:15.0:Update/libqt5-qtbase

expected origin is 'SUSE:SLE-15:GA' (changed)

no devel project found for openSUSE:Leap:15.0:Update/libqt5-qtbase
[I] 630848 ignored
```

**Comment by jberry-suse** [comment]
Just need `leaper` and `factory-auto` added as reviewers on `openSUSE:Maintenance`.

**Comment by andreasstieger** [comment]
<code>factory-auto</code> added. <code>leaper</code> needs discussion.

**Comment by jberry-suse** [comment]
Created `osrt-leaper-review-maintenance` and `osrt-leaper-review-maintenance.timer` by adapting `osrt-leaper-review` and put in comment only mode.

**Comment by lnussel** [comment]
The comment format for the leaper bot probably needs some tweaking to be readable and useful to maintenance.

**Comment by lnussel** [comment]
Also, maintenance may want #1607

**Comment by andreasstieger** [comment]
For <code>factory-source</code>, the comment in [SR#631299](https://build.opensuse.org/request/show/631229) does not make sense under the circumstances:

`the package needs to be accepted in openSUSE:Factory or openSUSE:Leap:15.0:Update or openSUSE:Leap:15.0 or openSUSE:Leap:15.0:Update or openSUSE:Leap:15.0 first
`

This is probably due to the <code>openSUSE:Backports:SLE-15:Update target. A maintenance update will frequently update multiple targets in parallel.

**Comment by lnussel** [comment]
factory-source would be obsoleted by leaper basically

**Comment by jberry-suse** [comment]
@lnussel Yeah, it may be time to include a bit of markdown in the reports.

@andreasstieger, Yes, `maintbot` and `factory-source` become `leaper`.

The most interesting report I've seen so far is https://build.opensuse.org/request/show/631199.

**Comment by jberry-suse** [comment]
I would also wager the deduplicate code is detrimental for multi-action requests.

**Comment by jberry-suse** [comment]
Yeah, the comment for 631199 becomes:

```
home:AndreasStieger:branches:OBS_Maintained:datovka/datovka.openSUSE_Backports_SLE-15_Update@6bd58ad6e6db0b0b17d11fbe5eba79f2 -> openSUSE:Backports:SLE-15:Update/datovka

expected origin is 'openSUSE:Leap:15.0' (changed)

home:AndreasStieger:branches:OBS_Maintained:datovka/datovka.openSUSE_Leap_15.0_Update@3b4c563a49635b9eff923c2c96b98326 -> openSUSE:Leap:15.0:Update/datovka

expected origin is 'openSUSE:Factory' (changed)

sr#629923 waiting for review by legal-auto

found pending submission against origin (openSUSE:Factory)

home:AndreasStieger:branches:OBS_Maintained:datovka/datovka.openSUSE_Leap_42.3_Update@8d175eea3836102cf9b01c4705c52641 -> openSUSE:Leap:42.3:Update/datovka

expected origin is 'openSUSE:Leap:42.2' (changed)

home:AndreasStieger:branches:OBS_Maintained:datovka/libisds.openSUSE_Backports_SLE-15_Update@53eba75acdee0ccd637dc9a14ee33f4d -> openSUSE:Backports:SLE-15:Update/libisds

expected origin is 'openSUSE:Leap:15.0' (changed)

home:AndreasStieger:branches:OBS_Maintained:datovka/libisds.openSUSE_Leap_15.0_Update@a86cddd25b67cdafbd88fa3d17b0192a -> openSUSE:Leap:15.0:Update/libisds

expected origin is 'openSUSE:Factory' (changed)

sr#628856 waiting for review by legal-auto

found pending submission against origin (openSUSE:Factory)

home:AndreasStieger:branches:OBS_Maintained:datovka/libisds.openSUSE_Leap_42.3_Update@ff38519231dffd4651ef753f0c3638ab -> openSUSE:Leap:42.3:Update/libisds

expected origin is 'openSUSE:Leap:42.2' (changed)
```

I'll open PR.

**Comment by jberry-suse** [comment]
So far haven't seen any rejections except https://build.opensuse.org/request/show/631967 by factory-auto which is correct based on the rules. Seems like both required reviews are operating properly. The leaper reviews seem accurate, but not sure what maintenance thoughts on usefulness.

**Comment by andreasstieger** [comment]
The Leaper reviews w.r.t. submission source frequently do not make sense:

In Factory or Leap development, the package is expected to be submitted repeatedly from the same source, e.g. a development project or a SLE parent tree. This is not the case for openSUSE Leap maintenance. In fact, they can be submitted from any source, usually a maintenance branch specific for this update. The usual scenarios here is backporting fixes to the stable releases, or making major version bumps in Factory while doing only minor ones in Leap.

So the message that the bot leaves is frequently off. The only comparison that is relevant is if a package that is supposed to come from SLE is submitted in a way that may cause it to diverge. Everything else is moot for Maintenance from what I can see. Possibly the message could be adjusted for this case.

**Comment by jberry-suse** [comment]
Need to revert #1733 once this is done

**Comment by jberry-suse** [comment]
Reviving an old discussion as #2126 is in and should allow for easily approving "forks" / workarounds via the review and tracking them as such. Using the new group, [origin-reviewers-maintenance](https://build.opensuse.org/group/show/origin-reviewers-maintenance), we could go ahead and try out origin-manager on the maintenance workflow for `openSUSE:Leap:15.2[:NonFree]` as I have already created the `OSRT:OriginConfig` attributes. All that should be needed (assuming gocd updates automatically to latest code @coolo) is to add the `origin-manager` user as a reviewer on `openSUSE:Maintenance`. Any action targeting a project without an `OSRT:OriginConfig` (all, but Leap 15.2) will be ignored. Any origin changes will be annotated and a human sign-off requested via the new group (indicated above).

Assuming this works well enough I would suggest we disabled `leaper` and `maintbot` as was the original goal of all this.
